src/components/button/button.tsx

```diff
@@ -33,37 +35,46 @@ export const Button = React.forwardRef<any, ButtonProps>(
       tabIndex,
       variant = 'default',
       color,
+      isLoading,
       ...props
     }: ButtonProps,
     ref,
   ) => {
     const theme = useTheme();
     const { mode } = useMode();
 
+    const isDisabled = disabled || isLoading;
+
     return (
       <StyledButton
         as={component === 'link' ? Link : component}
         type={type}
-        disabled={disabled}
-        className={cx(disabled && 'disabled', iconLeft && 'iconLeft', iconRight && 'iconRight', className)}
+        disabled={isDisabled}
+        className={cx(isDisabled && 'disabled', iconLeft && 'iconLeft', iconRight && 'iconRight', className)}
```

Comment:
you can add disabled styles in `StyledButton` using `:disabled` selector and then just add disabled or not